browser/components/newtab/aboutNewTabService.js

```diff
@@ -37,6 +36,17 @@ const PREF_MATCH_OS_LOCALE = "intl.locale.matchOS";
 // The preference that tells what locale the user selected
 const PREF_SELECTED_LOCALE = "general.useragent.locale";
 
+// The preference that tells what remote mode is enabled.
+const PREF_REMOTE_MODE = "browser.newtabpage.remote.mode";
+const MODE_CHANNEL_MAP = {
```

Comment:
Can you please move this mapping declaration into a JSM?

`NewTabRemoteResources.jsm` in `browser/components/newtab` should do.